fundamental-concepts/template-authorization/5_controllers.md

```diff
@@ -0,0 +1,46 @@
+The third UNIX access right is the right to execute an executable file, denoted by **x** in the `ls`
+output. While in UNIX the right to access is per file, in DAML the right to execute is more fine
+grained and denoted per choice of a contract. You give a party the right to execute a choice, by making
+it the `controller` of that choice.
+
+![controller](assets/controller.png)
+
+As in the case of `signatories` and `observers`, the `controller` can be a field of type `Party` of
+the contract template containing the choice. Different from `signatories` and `observers`, it can
+also be specified as a choice argument such as in the `SendMessage` choice:
+
+![flexible_controller](assets/flexible_controller.png)
+
+Lets confirm that the in-memory ledger enforces `controllers`. Append the line
+
+<pre class="file" data-target="clipboard">
+  c1 <- submit bob $ create User with username = bob, following = []
+  submit bob $ exercise c1 $ Follow with userToFollow = alice
+</pre>
+
+We create a new `User` contract for `Bob`. Then `Bob` exercises the `Follow` choice with `Alice` as
+argument. This passes, because `Bob` controls the `Follow` choice of his `User` contract and hence
+has the right to execute the choice.
+
+Now append
+
+<pre class="file" data-filename="daml/User.daml" data-target="append">
+  submit alice $ exercise c1 $ Follow alice
+</pre>
+
+to the scenario. You will get a scenario error:
```

Comment:
I get a different error

`Scenario execution failed on commit at User:54:3:
  Attempt to exercise a consumed contract #4:0 (User:User)
  Consumed by: #5:1`

This is what my scenario looks like for this step

`test = scenario do
  alice <- getParty "Alice"
  bob <- getParty "Bob"
  charlie <- getParty "Charlie"
  c0 <- submit alice $ create User with username = alice, following = []
  submit alice $ archive c0
  submit alice $ create User with username = alice, following = [bob, charlie]
  submitMustFail bob $ create User with username = alice, following = []
  c1 <- submit bob $ create User with username = bob, following = []
  submit bob $ exercise c1 $ Follow with userToFollow = alice
  submit alice $ exercise c1 $ Follow alice`

Reply:
I fixed this, thanks for spotting.